src/test/scala/com/gd/twitteranalytics/TweetsTransformerTest.scala

```diff
@@ -0,0 +1,59 @@
+package twitteranalytics
+
+import java.sql.Timestamp
+import java.util.Date
+
+import com.gd.twitteranalytics.TweetsTransformer
+import com.holdenkarau.spark.testing.{DataFrameSuiteBase, StreamingSuiteBase}
+import org.scalatest.FreeSpec
+import twitter4j.{TwitterObjectFactory}
+import org.scalamock.scalatest.MockFactory
+
+import scala.io.Source
+
+class TweetsTransformerTest extends FreeSpec with StreamingSuiteBase with MockFactory  with DataFrameSuiteBase {
+
+  "TransformTweets" - {
+    val constDate = new Date("Wed Sep 26 10:37:20 PDT 2018")
+    val text = "This is the #Sample #text for #test purpose"
+    val origDate = new Timestamp(constDate.getTime)
+
+    "getDateAndText function" - {
+    "when presented with a complete tweet status" - {
+      "should retrieve date and text from the tweets status" in {
+        val rawJson = Source.fromURL(getClass.getResource("/tweetStatus.json")).getLines.mkString
+        val tweetStatus = TwitterObjectFactory.createStatus(rawJson)
+        val input = List(List(tweetStatus))
+        val expected = List(List((constDate,text)))
+        testOperation(input,TweetsTransformer.getDateAndText,expected,ordered = false)
+      }
+    }
+  }
+    "getOutputAsDataFrame function" - {
+      "when presented with an RDD of Date and String" - {
+        "should generate a Dataframe of columns event_date[TimeStamp] and text[String]" in {
+          val sqlCtx = sqlContext
+          import sqlCtx.implicits._
+
+          val expectedDf = Seq((origDate,text),(origDate,text)).toDF("event_date","text")
+          val inputRdd = sc.parallelize(Seq((constDate, text), (constDate, text)))
+          val generatedDf = TweetsTransformer.convertRddToDataFrame(inputRdd)
+          assertDataFrameEquals(expectedDf,generatedDf)
+        }
+      }
+    }
+    "updateDateColFormat function" - {
+      "when presented with a DataFrame of format YYYY/MM/DD HH:MM:SS" -{
+        "should update the format to YYYYMMDD HH.MM" in {
+          val sqlCtx = sqlContext
+          import sqlCtx.implicits._
+
+          val sourceDf = Seq((origDate,text),(origDate,text)).toDF("event_date","text")
+          val generatedDf = TweetsTransformer.updateDateColFormat(sourceDf,"event_date")
+          val dateFromDf = generatedDf.select("event_date").first.getString(0)
+          assert("20180926 10.37" === dateFromDf)
```

Comment:
technically, it's not a date, but date time.
if we want to include time info, it should be stored in a subdirectory: event_date=.../event_hour=.../event_minute=
such partitioning schema will facilitate data lookup.

Reply:
I kept time info in Date for my daily testing purpose to make sure that partitioning is being done properly. In production of course, we can only keep eventDate. 
So, here shall I change to the suggested structure
event_date=.../event_hour=.../event_minute=
 or remove the time and only keep Date

Reply:
> In production of course, we can only keep eventDate.

I think it rather depends on incoming data volume - in case we are receiving 100k+ events per second, it may make sense to introduce a more granular partitioning than a date-based.

In our case, for the sake of simplicity, date-based partitioning is enough. 

> I kept time info in Date for my daily testing purpose to make sure that partitioning is being done properly.

reminder: I'd like to see a test for it, validating output structure.